src/pulumi_datarobot_utils/common/feature_flags.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2025 DataRobot, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pathlib
+from typing import Iterable
+
+import datarobot as dr
+import pulumi
+import yaml
+
+# TODO: Introduce custom types to avoid complex constructions like dict[tuple[...]...]
+
+def get_statuses(flags: Iterable[str]) -> dict[str, bool]:
+    client = dr.client.get_client()
+    flags_json = {"entitlements": [{"name": flag} for flag in flags]}
+    response = client.post("entitlements/evaluate/", json=flags_json)
```

Comment:
Looks like a gap in our Python Client - grepped the codebase and didn't see `entitlements`

We should get the responsible team to contribute to the client IMO